test/reserveSetup.js

```diff
@@ -0,0 +1,409 @@
+const ConversionRatesV2 = artifacts.require("MockEnhancedStepFunctions.sol");
+const ConversionRatesV1 = artifacts.require("MockConversionRate.sol");
+const KyberFprReserveV2 = artifacts.require("KyberFprReserveV2");
+
+const Helper = require("./helper.js");
+const BN = web3.utils.BN;
+
+//global variables
+//////////////////
+const {precisionUnits, ethAddress, zeroAddress, 
+    zeroBN} = require("./helper.js");
+
+//block data
+let validRateDurationInBlocks = 1000;
+
+// imbalance data
+let minimalRecordResolution = new BN(2);
+let maxPerBlockImbalance = 40000;
+let maxTotalImbalance = maxPerBlockImbalance * 12;
+
+module.exports.setupConversionRateV1 = async function(tokens, admin, operator, alerter, needListingToken) {
+    let convRatesInst = await ConversionRatesV1.new(admin);
+
+    //set pricing general parameters
+    await convRatesInst.setValidRateDurationInBlocks(validRateDurationInBlocks);
+
+    let baseBuyRate = [];
+    let baseSellRate = [];
+    let compactBuyArr = [];
+    let compactSellArr = [];
+
+    let qtyBuyStepX = [0, 5000, 10000, 16000, 28000, 32000, 45000];
+    let qtyBuyStepY = [ 0, 10, 20, 32, 44, 56, 101];
+    let qtySellStepX = [0, 5000, 11000, 18000, 30000, 36000, 48000];
+    let qtySellStepY = [ 0, 12, 22, 33, 45, 58, 110];
+    let imbalanceBuyStepX = [0, 5000, 15000, 21000, 28000, 33000, 45000];
+    let imbalanceBuyStepY = [0, -110, -160, -250, -440, -1000, -1600];
+    let imbalanceSellStepX = [-48000, -39000, -25000, -19000, -14000, -10000, 0];
+    let imbalanceSellStepY = [-1500, -1200, -440, -320, -200, -75, 0];
```

Comment:
I see in some tests its with 10 steps
Are steps configured in a different place in the js file?

Reply:
for gas consumption v1 tests, the number of steps reported are total of quantity and imbalance steps. For v2, it is only imbalanace step

Reply:
Ahhh got it.
OK